### PR TITLE
Fix deploy script, local dev in Docker

### PR DESCRIPTION
**Pull request by haradan**
Updated the `bin/deploy` script to use the built bundler for rake tasks.

Prevented overwriting the built bundler when mounting the local repository into Docker for local development.

### CODE REVIEW

**Review by leukosaima — APPROVED** [review]

File: db/schema.rb
Comment (plural) [inline db/schema.rb:1]: why is this file included in the PR?

File: README.md [hunk 1]
```diff
@@ -102,10 +102,11 @@ echo "RAILS_ENV=development" >> .env
 - Deploy the app
 
 ```shell
-docker compose -f .\docker-compose.yml -f .\docker-compose.override.init.yml up -d
+bin/deploy
```
Comment (plural) [inline README.md:105]: not needed in this PR, but the deploy shell script isn't deploying anything really - it's building and preparing. The name's a bit misleading.

File: docker-compose.override.yml [hunk 2]
```diff
@@ -5,6 +5,8 @@ services:
   app:
     ports:
       - "3000:3000"
+    volumes:
```
Comment (plural) [inline docker-compose.override.yml:8]: how does this interact with the copied set from the Dockerfile?

File: db/schema.rb
Comment (haradan) [reply db/schema.rb:1]: I'm not sure, I thought I ran the DB migration when I upgraded everything but when I ran it to test these changes it did this.

File: docker-compose.override.yml (hunk 2 above)
Comment (haradan) [reply docker-compose.override.yml:8]: It replaces everything in that directory with your local repository, just like it did before the Dockerfile changes.

File: README.md (hunk 1 above)
Comment (haradan) [reply README.md:105]: I'm not sure what you mean, this is the script we run on the droplet to deploy the app.

File: db/schema.rb
Comment (plural) [reply db/schema.rb:1]: ok, but would you please remove it from this PR?  It's unrelated.

File: docker-compose.override.yml (hunk 2 above)
Comment (plural) [reply docker-compose.override.yml:8]: i guess i'm asking why this is in the override and if it will clobber the work of the main Dockerfile when we want to use that.  docker-compose.override.yml is used by default unless you specify a specify list of files to use.
